Железнов А.М. Лаб. 4 Группа 6312

Artem362a opened the pull request:
**ФИО:** Железнов Артём
**Номер группы:** 6312 
**Номер лабораторной:** 4  
**Краткое описание исходного кода:** Я взял код с https://github.com/marcoemrich/Refactoring-Katas/blob/master/Yahtzee/python/yahtzee.py
**Краткое описание проделанной работы:** Я сделал рефакторинг этого кода.

IlyaOv commented:
Проделана важная работа, но первая лабораторная работа направлена на **стандарты кодирования**. Прочитайте, пожалуйста, условия ее сдачи.

derasoft004 commented:
> Проделана важная работа, но первая лабораторная работа направлена на **стандарты кодирования**. Прочитайте, пожалуйста, условия ее сдачи.

Я привел свой старый код к  **стандартам кодирования** PEP8, также структурировал плохо собранный проект, применил **Осмысленное название коммитов**, в каждом коммите исправлял разные аспекты (может слегка больше чем один), **Размер моего исходного кода блолее 300 строк**, применено 5 различных правил оформления кода: импорты  приведены к стандарту PEP;  добавлены нужные отступы по PEP8; добавлены файлы 'requirements.txt' и '.env' (а так же 'config.py'), что относится к правильному оформлению кода, так как секретные ключи и токены не должны содержаться в коде; повторяющиеся строки заменены более компактными выражениями; код избавлен от множества глобальных переменных, которые в этой области памяти находиться не должны. **В описании пул-риквеста ссылку на исходный источник редактируемого кода** указал. 
Что мне нужно поправить, добавить? Подскажите, пожалуйста.

derasoft004 commented:
ой там не type-scripts а type-hints 😃

IlyaOv reviewed (changes requested):
Если приводится к PEP 8, то из вашего ответа и коммитов я пока увидел указание на 3 аспекта:
- оформление импортов
- отступы
- type hints

Их проверил, замечания написал. 

Остальное безусловно является хорошей практикой и естественно не возбраняется. (чему и посвящены другие лабораторные работы). Но, на мой взгляд, не относится непосредственно к PEP 8. Не утверждаю, что прав, приведите пример из документации, если ошибаюсь. Если это из других стандартов кодирования, укажите их.

Пока мне видится необходимость выбрать и поправить еще пару аспектов из PEP 8.

derasoft004 commented:
Исправил импорты по Вашему замечанию, поправил некоторые ветки с иф и условием в одной строке, поправил слишком длинные строки

IlyaOv commented:
Зачтено

Artem362a committed:
add source_code

add code for refactoring

Artem362a committed:
Убрал повторение кода в функциях ones,twos... 

Для того, чтобы сохранить изначальную логику класса (вдруг этот класс где-то еще используется), я оставил функции ones,twos...

Artem362a committed:
Добавил метод counts, немного поменял методы и убрал конструктор

Конструктор тут не нужен, т.к. он не используется в статических методах

Artem362a committed:
Убрал захардкоженные значения

Artem362a committed:
Добавил докстринги и привел код к PEP8

alxmcs commented on source_code.py:
А почему у файла нет расширения?

alxmcs commented on source_code.py:
У тебя каждый метод в классе по итогам рефакторинга получился статическим. Это точно качественный рефакторинг? Может быть, имеет смысл выделить больше одного класса? Или вообще отказаться от ООП и оставить все в виде процедур?

Artem362a committed:
Добавил расширение файлу

Artem362a committed:
Убрал класс

Я провел такой, мягкий рефакторинг. Все функции на месте, для того, чтобы логика работы с модулем сильно не поменялась

Artem362a commented:
Есть вопрос: в коде, для которого я проводил рефакторинг есть много функций (ones,twos,,....), у которых примерно похожая логика. Надо ли мне от них избавиться, или можно оставить все в таком виде? В своем рефакторинге я исходил из того, что код может быть уже где-то используется и сохранил эти функции, но я понимаю, что по-хорошему эти функции бы удалить, т.к. они просто засоряют код

alxmcs reviewed (approved):
Методы, конечно, не самые красивые, но, поскольку они представляют из себя комбинации в игре, сгодится и так.
